pkg/oci/oci.go

```diff
@@ -98,15 +117,48 @@ func (o *Oci) Push(ref string) (digest.Digest, error) {
 
 	delete(refDescriptor.Annotations, "org.opencontainers.image.ref.name")
 
-	pushDescriptor, err := oras.Push(o.ctx,
+	allowedMediaTypes := []string{
+		"application/vnd.oci.image.manifest.v1+json",
+		"application/octet-stream",
+		"application/vnd.oci.image.config.v1+json",
+		"application/vnd.unknown.config.v1+json",
+		"application/vnd.oci.image.layer.v1.tar+gzip",
+		"application/vnd.oci.image.layer.v1.tar",
+	}
+
+	opts := []oras.CopyOpt{oras.WithAllowedMediaTypes(allowedMediaTypes)}
+
+	memoryStore := content.NewMemory()
+	manifest, manifestdesc, err := content.GenerateManifest(nil, refDescriptor.Annotations, refDescriptor)
+	if err != nil {
+		return "", err
+	}
+
+	err = memoryStore.StoreManifest(ref, manifestdesc, manifest)
+	if err != nil {
+		return "", err
+	}
+
+	pushDescriptor, err := oras.Copy(o.ctx,
+		o.ociStore,
+		ref,
 		resolver,
+		"",
+		opts...)
+
+	if err != nil {
+		return "", errors.Wrap(err, "oras push tarball failed")
+	}
+
+	pushDescriptor, err = oras.Copy(o.ctx,
```

Comment:
A small comment on top of these copy operations would help understand the logic behind pushing the manifest and the tarball  separtely